build for KW40z, KW41z fails to generate isr_tables

**Describe the bug**
Building an example for a frdm_kw41z results in a failure in generating isr_tables when gpio portb is enabled:
```
[ 97%] Generating isr_tables.c
gen_isr_tables.py: multiple registrations at table_index 31 for irq 31 (0x1f)
Traceback (most recent call last):
  File "/[..]/zephyr/arch/common/gen_isr_tables.py", line 308, in <module>
    main()
  File "/[..]/zephyr/arch/common/gen_isr_tables.py", line 300, in main
    error("multiple registrations at table_index %d for irq %d (0x%x)" % (table_index, irq, irq))
  File "/[..]/zephyr/arch/common/gen_isr_tables.py", line 35, in error
    raise Exception()
Exception
make[2]: *** [zephyr/CMakeFiles/kernel_elf.dir/build.make:63: zephyr/isr_tables.c] Error 1
make[1]: *** [CMakeFiles/Makefile2:314: zephyr/CMakeFiles/kernel_elf.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

**To Reproduce**
Steps to reproduce the behavior:
1. enable GPIO_MCUX_PORTB in boards/arm/frdm_kw41z/Kconfig.defconfig
2. build e.g. hello_world example
3. cmake -DBOARD=frdm_kw41z
4. make
5. See error

**Impact**
Examples for frdm_kw41z or boards based on kw40z and kw41z will fail in building when all gpios are activated

**Screenshots or console output**
in debug mode, I'm getting this:
```
[ 97%] Generating isr_tables.c
gen_isr_tables.py: (32, 0)
gen_isr_tables.py: Configured interrupt routing
gen_isr_tables.py: handler    irq flags param
gen_isr_tables.py: --------------------------
gen_isr_tables.py: 0x1957     31  0     0x20000ed0
gen_isr_tables.py: 0x1957     31  0     0x20000edc
gen_isr_tables.py: 0x1957     30  0     0x20000ee8
gen_isr_tables.py: offset is 0
gen_isr_tables.py: num_vectors is 32
gen_isr_tables.py: multiple registrations at table_index 31 for irq 31 (0x1f)
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: 0f3bd0f43bf5bbc559303dc91d4fc3aacb58281e

**Additional context**
This check was recently introduced by #11273 by @andrewboie, issue is similar to k64 in #11275.
possible fix: Assign a different interrupt number in nxp_kw40z.dtsi and nxp_kw41z.dtsi.
How were the numbers 30/31 selected? Which number to select instead of 31?

--- a/dts/arm/nxp/nxp_kw40z.dtsi
+++ b/dts/arm/nxp/nxp_kw40z.dtsi
@@ -170,7 +170,6 @@
 		gpiob: gpio@400ff040 {
 			compatible = "nxp,kinetis-gpio";
 			reg = <0x400ff040 0x40>;
-			interrupts = <31 2>;
 			label = "GPIO_2";
 			gpio-controller;
 			#gpio-cells = <2>;

--- a/dts/arm/nxp/nxp_kw41z.dtsi
+++ b/dts/arm/nxp/nxp_kw41z.dtsi
@@ -173,7 +173,6 @@
 		gpiob: gpio@400ff040 {
 			compatible = "nxp,kinetis-gpio";
 			reg = <0x400ff040 0x40>;
-			interrupts = <31 2>;
 			label = "GPIO_1";
 			gpio-controller;
 			#gpio-cells = <2>;

